File: Producto/backend/src/main/java/backend/com/comercial/application/dto/ItemNVDTO.java
package backend.com.comercial.application.dto;

import backend.com.comercial.domain.enums.TipoItem;
import lombok.Data;
import java.math.BigDecimal;
import java.util.List;

@Data
public class ItemNVDTO {

    /**
     * Id del ítem ya persistido (al editar una NV existente). Null para ítems nuevos —
     * permite distinguir "ítem que ya tenía una OP asociada" de "ítem nuevo sin OP".
     */
    private Long idItemNV;
    private Integer nroItem;
    private Integer articuloId;
    private Integer cantidad; // Restaurado
    private String modelo;
    private String tela;
    private String composicion;
    private String color;
    private String talla;
    private String genero;
    private Long proveedorId;
    private String llevaLogo;
    private String logoDetalle;
    private TipoItem itemType;
    /**
     * Costeo existente (APROBADO) a vincular manualmente a la OP de este ítem, cuando
     * es un ítem tipo OP nuevo que no hereda un costeo desde la EVN plantilla.
     */
    private Long costeoId;
    private Boolean requiereOt;
    private String detalleOt;
    private List<TallaDTO> tallas;
    private BigDecimal precioUnitario;
    private BigDecimal total;

    @Data
    public static class TallaDTO {
        private String talla;
        private Integer cantidad;
    }
}

File: Producto/frontend/src/remote/DTO/ItemNVDTO.js
/**
 * DTO para representar un Item de Nota de Venta.
 * Basado en ItemNV.java
 */
export class ItemNVDTO {
    constructor(data = {}) {
        this.idItemNV = data.id || data.idItemNV || null;
        this.nroItem = data.nroItem || null;
        this.productoId = data.productoId || null;
        this.proveedorId = data.proveedorId || null;
        this.cantidad = data.cantidad || 0;
        this.precioUnitario = data.precioUnitario || 0;
        this.modelo = data.modelo || '';
        this.tela = data.tela || '';
        this.composicion = data.composicion || '';
        this.color = data.color || '';
        this.talla = data.talla || '';
        this.genero = data.genero || '';
        this.llevaLogo = data.llevaLogo || 'N/A';
        this.logoDetalle = data.logoDetalle || '';
        this.tallas = data.tallas || [];
        this.itemType = data.itemType || data.tipoItem || '';
        this.requiereOt = data.requiereOt || data.generaOt || false;
        this.detalleOt = data.detalleOt || '';
        this.currency = 'CLP';
        this.totalItem = data.total || data.totalItem || 0;
    }
}
